src/components/cc-pricing-header/cc-pricing-header.js

```diff
@@ -78,73 +119,119 @@ export class CcPricingHeader extends LitElement {
       .replace('$US', '$');
   }
 
+  /**
+   * Returns the localized string corresponding to a given temporality type.
+   *
+   * @param {Temporality['type']} type - the temporality type
+   * @return {string} the localized string corresponding to the given temporality type
+   */
+  _getPriceLabel (type) {
+    if (type === 'second') {
+      return i18n('cc-pricing-header.price-name.second');
+    }
+    if (type === 'minute') {
+      return i18n('cc-pricing-header.price-name.minute');
+    }
+    if (type === 'hour') {
+      return i18n('cc-pricing-header.price-name.hour');
+    }
+    if (type === '1000-minutes') {
+      return i18n('cc-pricing-header.price-name.1000-minutes');
+    }
+    if (type === 'day') {
+      return i18n('cc-pricing-header.price-name.day');
+    }
+    if (type === '30-days') {
+      return i18n('cc-pricing-header.price-name.30-days');
+    }
+  }
+
+  /**
+   * Retrieves the currency corresponding to the selected currency code.
+   * Dispatches a `cc-pricing-header:change-currency` event with the currency as its payload.
+   *
+   * @param {Event} e - the event that called this method
+   */
   _onCurrencyChange (e) {
-    const currency = this.currencies.find((c) => c.code === e.target.value);
+    const currency = this.currencyList.find((c) => c.code === e.target.value);
     dispatchCustomEvent(this, 'change-currency', currency);
   }
 
+  /**
+   * Retrieves the temporality corresponding to the selected temporality type.
+   * Dispatches a `cc-pricing-header:change-temporality` event with the temporality as its payload.
+   *
+   * @param {Event} e - the event that called this method
+   */
+  _onTemporalityChange (e) {
+    const temporality = this.temporalityList.find((t) => t.type === e.target.value);
+    dispatchCustomEvent(this, 'change-temporality', temporality);
+  }
+
+  /**
+   * Retrieves the zone id from the event payload.
+   * Dispatches a `cc-pricing-header:change-zone` event with the zone id as its payload.
+   *
+   * @param {Event} e - the event that called this method
+   */
   _onZoneChange (e) {
-    const zoneName = e.target.value;
-    dispatchCustomEvent(this, 'change-zone', zoneName);
+    const zoneId = e.target.value;
+    dispatchCustomEvent(this, 'change-zone', zoneId);
   }
 
   willUpdate (changedProperties) {
-    if (changedProperties.has('zones')) {
-      this._sortedZones = sortZones(this.zones);
+    if (changedProperties.has('zoneList') && Array.isArray(this.zoneList.value)) {
+      this._sortedZoneList = sortZones(this.zoneList.value);
     }
   }
```

Comment:
suggestion: I would also consider removing the `willUpdate` and move the code to the render function, we need to discuss this